elide-core/src/main/java/com/yahoo/elide/core/PersistentResource.java

```diff
@@ -1267,14 +1269,52 @@ public Resource toResource(final Supplier<Map<String, Relationship>> relationshi
             } else {
                 data = new Data<>(resources);
             }
-            // TODO - links
-            relationshipMap.put(field, new Relationship(null, data));
+
+            Map<String, String> relationshipLinks = getRelationshipLinks(field);
+            relationshipMap.put(field, new Relationship(relationshipLinks, data));
         }
 
         return relationshipMap;
     }
 
     /**
+     * Get resource level self link.
+     * @return self link of current resource
+     */
+    protected Map<String, String> getResourceLevelLink() {
+        return ImmutableMap.of("self", getResourceUrl());
+    }
+
+    /**
+     * Get relationship links based for current resource.
+     * @param field relationship field
+     * @return links for relationship object
+     */
+    protected Map<String, String> getRelationshipLinks(final String field) {
+        String resourceUrl = getResourceUrl();
+        return ImmutableMap.of(
+                "self", String.join("/", resourceUrl, "relationships", field),
+                "related", String.join("/", resourceUrl, field));
+    }
+
+    /**
+     * Get base resource url of current resource.
+     * @return resource url
+     */
+    protected String getResourceUrl() {
+        StringBuilder result = new StringBuilder();
+        for (String name : lineage.getKeys()) {
+            if (name.equals(getType())) {
+                break;
+            }
+            PersistentResource value = lineage.getRecord(name).get(0);
```

Comment:
I actually don't think this right.  We're using the persistent resource's _type_ rather than the _relationship name_ to build the URL.  These links won't be correct.  

I'm wondering if a better approach is pass the URL through the request scope?  Either that or the lineage needs to be augmented to include the relationship names used to traverse to a given persistent resource.